### PR TITLE
OPEX-1054: Adds health tab documentation.

### PR DESCRIPTION
**Pull request by sharmoo**
This adds health tab documentation for logs streaming in connection with https://auth0team.atlassian.net/browse/OPEX-1054

### CODE REVIEW

File: articles/logs/streams/health.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+description: All Log Streams types let you view your stream health in the health tab.
+toc: false
+topics:
+ - logs
+ - streams
+ - event-streams
+contentType: how-to
+---
+
+# HTTP Event Log Streams
```
Comment (camicano) [inline articles/logs/streams/health.md]: not sure this part is necessary since the explanation of what event streams are is elsewhere.

File: articles/logs/streams/health.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+HTTP Event Log Streams let you export your log events to the server or target of your choice. When Auth0 creates a log entry for your tenant, a copy is automatically sent to a given URL via an HTTP POST request.
+
+If you use Amazon Web Services, Auth0 also offers an [AWS EventBridge integration](/integrations/aws-eventbridge).
+
```
Comment (camicano) [inline articles/logs/streams/health.md:15]: Before we dive into the how, maybe give a quick overview of what is log stream health.

# What is Log Stream Health

Log Stream Health is the functionality that allows customers to troubleshoot issues with Log Streams. It allows customers to get the status of the latest delivery attempts of each stream to make it easy to identify problems in the delivery.

# How to use Stream Health

** add some stuff here**

Status types:

| Status | Description |
|---------|-------------|
| Active (with no errors) | This means that the latest log deliveries have been successful |
| Active (with errors) | The log deliveries are active but the stream failed to deliver logs at one or more points |
| Paused | The customer has pausued the stream. |
| Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option "restart" |

File: articles/logs/streams/health.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+## Viewing an Existing Event Stream's Health
+
+1. Log in to the [Auth0 Dashboard](${manage_url}).
+2. Navigate to **Logs > Streams**.
+3. Click on any existing stream listed therein.
```
Comment (camicano) [inline articles/logs/streams/health.md]:
```suggestion
3. Click on a stream
```

File: articles/logs/streams/health.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+4. Select the **Health** tab. An explanation of the following potential states is as follows:
```
Comment (camicano) [inline articles/logs/streams/health.md]:
```suggestion
4. Select the **Health** tab

File: articles/logs/streams/health.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+| Status | Description |
+|---------|-------------|
+| Active (with no errors) | This means that the latest log deliveries have been successful |
+| Active (with errors) | The log deliveries are active but the stream failed to deliver logs at one or more points |
+| Paused | The customer has pausued the stream. |
+| Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option "restart" |
```
Comment (camicano) [inline articles/logs/streams/health.md]:
```
```

**Review by camicano — CHANGES_REQUESTED** [review]
In general I would focus more of functionality since this is a how-to doc. @tony-jones might able to help give better explanation of the user flow 😄

File: articles/logs/streams/health.md (hunk 5 above)
Comment (erjohnson) [reply articles/logs/streams/health.md]: Maybe change:

`This means that the latest log deliveries have been successful`

to

`The latest log deliveries have been successful` or `The latest log deliveries were successful`

File: articles/logs/streams/health.md (hunk 1 above)
Comment (erjohnson) [inline articles/logs/streams/health.md]: The `#` (h1) should match the title: `Log Streams Health`

File: articles/logs/streams/health.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+3. Click on a stream.
+4. Select the **Health** tab. 
+
+## Understanding Stream Health States
+
+**add some stuff here**
+
+Status types:
+
+| Status | Description |
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:29]: I think these should be described in a user friendly way in terms of scenarios. Maybe like what we've been talking about in slack?

https://auth0.slack.com/archives/CEJHNE5QV/p1585588481287000?thread_ts=1585584951.281100&cid=CEJHNE5QV

File: articles/logs/streams/health.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+### Errors
+
+Errors are populated in the Logs Health Table when logs could not be delivered with corresponding error codes and counter information. TBD here is how to use counter information to recover logs and the time frame for recovery TBD.
+
+### Pause/Resume Streams
+
+You may pause your stream by using the drop down option menu in the log stream list view and clicking 'pause.' Streams may be resumed by returning to the dropdown and clicking 'resume.'
+
+If a stream is paused, TBD here is how to recover your logs during the pause time frame and the time frame for recovery TBD.
+
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:55]: We could add a screenshot for pause/resume

File: articles/logs/streams/health.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+### Suspensions
+
+If your server is unable to receive the stream, we will retry up to fifty times to deliver the event; otherwise, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' TBD here is how to recover logs during hte suspension state  and and the time frame for recoveryTBD (if all of the above recovery methods are the same, perhaps put it under subheading resolving errors and undelivered logs during paused, suspended, or active states  )
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md]: screenshots would be helpful here too

Comment (Deborah-Digges) [inline articles/logs/streams/health.md]: Let's not make a gaurantee of the number `50`. That's an internal implementation detail we don't want to expose to customers

File: articles/logs/streams/health.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
+---
+title: Log Streams Health
+description: All Log Streams types let you view your stream health in the health tab.
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:3]: This sentence could be better structured as "View the health for any type of log stream in the health tab"

File: articles/logs/streams/health.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+**Stream Health has a variety of different states. These status types are described below.**
+
+Status types:
+
+| Status | Description |
+|---------|-------------|
+| Active (with no errors) | The latest log deliveries were successful |
+| Active (with errors) | The log deliveries are active but the stream failed to deliver logs at one or more points. |
+| Paused | The customer has paused the stream |
+| Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option `restart` |
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:34]:
```suggestion
| Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to the list view and clicking the dropdown option `restart` |
```

Comment (Deborah-Digges) [inline articles/logs/streams/health.md:34]: The dashboard term used is `disabled`:

```suggestion
| Disabled | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option `restart` |
```

File: articles/logs/streams/health.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+
+## Streaming delivery errors, pauses, or suspensions. 
+
+Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. If a stream is paused or suspended, log deliveries will resume from the last delivery point upon stream resume or restart. Please note, however, that logs are only retained up to the retention period as noted on your account. Once the retention period is up, missing logs will not be available for retreival.
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:39]: `as it is triggered in our system` suggests that we attempt delivery in real-time and we make no such promises. We should reword it to avoid making that promise inadvertently

Comment (Deborah-Digges) [inline articles/logs/streams/health.md:39]:
```suggestion
Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. If a stream is paused or disabled, log deliveries will resume from the last delivery point when the stream is resumed or restarted. Please note, however, that logs are only retained up to the retention period as noted on your account. Once the retention period is up, missing logs will not be available for retrieval. 
```

File: articles/logs/streams/health.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+| Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option `restart` |
+
+
+## Streaming delivery errors, pauses, or suspensions. 
+
+Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. If a stream is paused or suspended, log deliveries will resume from the last delivery point upon stream resume or restart. Please note, however, that logs are only retained up to the retention period as noted on your account. Once the retention period is up, missing logs will not be available for retreival. 
+
+### Errors
+
+Errors (from the last 24 hours only) are populated in the Logs Health Table when logs could not be delivered. The log errors are populated with corresponding error codes and checkpoint information. Additional information on retrieving associated logs via checkpoint can be found here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint.
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:43]: We may or may not want to tell them about this, because as to @machuga 's point, it is not possible to replicate the exact payload body using the startingCursor alone. More details here:

https://auth0.slack.com/archives/CEJHNE5QV/p1585847510013100

Let's remove this line. We might update the UI to remove the `copy to clipboard` action for the starting cursor. TBD

File: articles/logs/streams/health.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+### Errors
+
+Errors (from the last 24 hours only) are populated in the Logs Health Table when logs could not be delivered. The log errors are populated with corresponding error codes and checkpoint information. Additional information on retrieving associated logs via checkpoint can be found here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint. 
+
+### Pause/Resume Streams
+
+Perhaps you need to pause streaming for a while during an upgrade or error resolution?
+
+You may pause your stream by using the drop down option menu in the log stream list view and clicking 'pause.' Streams may be resumed by returning to the dropdown and clicking 'resume.'
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:49]:
```suggestion
You may pause your stream by using the drop-down option menu in the log stream list view and clicking `Pause Stream'. Streams may be resumed by returning to the drop-down and clicking 'Resume Stream'.
```

File: articles/logs/streams/health.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+![Pausing a Stream](/media/articles/logs/streams-health/stream-pause.png "Pausing a Stream")
+
+
+If a stream is paused, you may resume it any time using the same steps.
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:54]: Is this line needed? Didn't we just explain how to resume a stream in the lines above?

File: articles/logs/streams/health.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+You may pause your stream by using the drop down option menu in the log stream list view and clicking 'pause.' Streams may be resumed by returning to the dropdown and clicking 'resume.'
+
+![Pausing a Stream](/media/articles/logs/streams-health/stream-pause.png "Pausing a Stream")
+
+
+If a stream is paused, you may resume it any time using the same steps. 
+
+### Suspensions
+
+If your server is unable to receive the stream, we will retry  to deliver the event; if the events cannot be received after several consecutive retries, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.'
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:58]:
```suggestion
If we receive an error while attempting to deliver events to your log stream, we will retry event delivery; if the events cannot be received after several consecutive retries, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the drop-down menu to click `Restart Stream' 
```

File: articles/logs/streams/health.md [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+![Pausing a Stream](/media/articles/logs/streams-health/stream-pause.png "Pausing a Stream")
+
+
+If a stream is paused, you may resume it any time using the same steps. 
+
+### Suspensions
+
+If your server is unable to receive the stream, we will retry  to deliver the event; if the events cannot be received after several consecutive retries, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' 
+
+### Resolving Errors and Undelivered Logs During Paused, Active, or Suspsended States
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:60]:
```suggestion
### Resolving Errors during Paused, Active, or Suspended States
```

File: articles/logs/streams/health.md [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+
+Don't see logs on your end? Use the error table to find out why. We now log recent errors and their descriptions to help you investigate issues with your stream.
+
+![Disabled Stream](/media/articles/logs/streams-health/disabled-stream.png "A Disabled Stream")
+
+1. On a stream's **Health** tab: please reference any errors and the associated counter information. 
+2. Follow the instructions outlined here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint
```
Comment (Deborah-Digges) [inline articles/logs/streams/health.md:67]: Again, we might remove this altogether.

File: articles/logs/streams/health.md [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+topics:
+ - logs
+ - streams
+ - event-streams
+contentType: how-to
+---
+
+# Viewing an Existing Event Stream's Health
+
+Log Stream Health is the functionality that allows customers to troubleshoot issues with Log Streams. It allows customers to get the status of the latest delivery attempts of each stream to make it easy to identify problems in the delivery.
```
Comment (camicano) [inline articles/logs/streams/health.md:14]:
```suggestion
Log Stream Health allows customers to troubleshoot issues with Log Streams. It doesn it by tracking the status of the latest delivery attempts of each stream to make it easy to identify problems in the delivery.
```

File: articles/logs/streams/health.md [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+
+## Viewing an Existing Event Stream's Health
```
Comment (camicano) [inline articles/logs/streams/health.md:16]:
```suggestion
## Viewing the Health of an Existing Log Stream
```

File: articles/logs/streams/health.md [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+
+1. Log in to the [Auth0 Dashboard](${manage_url}).
+2. Navigate to **Logs > Streams**.
+3. Click on a stream.
+4. Select the **Health** tab. 
+
+## Understanding Stream Health States
```
Comment (camicano) [inline articles/logs/streams/health.md:23]:
```suggestion
## Understanding Stream Health Status
```

I think is important to be consistent with the terms

File: articles/logs/streams/health.md (hunk 11 above)
Comment (camicano) [inline articles/logs/streams/health.md:39]:
```suggestion
Auth0 events are delivered to your server via a mechanism that sends batches of log events as they are triggered in our system. If a stream is paused or suspended, log event deliveries will resume from the last delivery point once the stream is resumes or restarted. Please note that log events are only retained up to the retention period as noted on your account. Once the retention period is up, logs will not longer be available for retrieval. 
```

File: articles/logs/streams/health.md (hunk 12 above)
Comment (camicano) [inline articles/logs/streams/health.md:43]:
```suggestion
In the event that a batch of logs was unable to be delivered an errors will be logged in the Logs Health Table for up to 24 hours after the delivery attempt. The errors will include an error code as well as checkpoint information. Additional information on retrieving associated logs via checkpoint can be found here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint. 
```

File: articles/logs/streams/health.md [hunk 21]
```diff
@@ -0,0 +1,67 @@
...
+
+Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. If a stream is paused or suspended, log deliveries will resume from the last delivery point upon stream resume or restart. Please note, however, that logs are only retained up to the retention period as noted on your account. Once the retention period is up, missing logs will not be available for retreival. 
+
+### Errors
+
+Errors (from the last 24 hours only) are populated in the Logs Health Table when logs could not be delivered. The log errors are populated with corresponding error codes and checkpoint information. Additional information on retrieving associated logs via checkpoint can be found here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint. 
+
+### Pause/Resume Streams
+
+Perhaps you need to pause streaming for a while during an upgrade or error resolution?
```
Comment (camicano) [inline articles/logs/streams/health.md:47]: I am not sure we need this question.

File: articles/logs/streams/health.md (hunk 13 above)
Comment (camicano) [inline articles/logs/streams/health.md:49]:
```suggestion
Streams can be paused by using the drop down option menu in the log stream list view and clicking 'pause.' Streams may be resumed by returning to the dropdown and clicking 'resume.'
```

All the docs are written in passive voice and I think this part should remain consistent with that. Also I am not sure if this should be documented, seems pretty basic. @tony-jones @erjohnson what do you think?

File: articles/logs/streams/health.md (hunk 14 above)
Comment (camicano) [reply articles/logs/streams/health.md:54]: agreed. not sure this whole section is needed TBH

File: articles/logs/streams/health.md (hunk 15 above)
Comment (camicano) [inline articles/logs/streams/health.md:58]:
```suggestion
In the event that the stream target is unable to receive a delivery attempt, our system to retry several times; if the events cannot be successfully delivered after several consecutive retries, we will suspend the stream. The event can be restarted manually by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' 
```

File: articles/logs/streams/health.md [hunk 22]
```diff
@@ -0,0 +1,67 @@
...
+
+If a stream is paused, you may resume it any time using the same steps. 
+
+### Suspensions
+
+If your server is unable to receive the stream, we will retry  to deliver the event; if the events cannot be received after several consecutive retries, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' 
+
+### Resolving Errors and Undelivered Logs During Paused, Active, or Suspsended States
+
+Don't see logs on your end? Use the error table to find out why. We now log recent errors and their descriptions to help you investigate issues with your stream.
```
Comment (camicano) [inline articles/logs/streams/health.md:62]:
```suggestion
The error table logs recent errors in the delivery of logs with a description description to help understand and troubleshoot issues with a stream.
```

File: articles/logs/streams/health.md (hunk 17 above)
Comment (camicano) [reply articles/logs/streams/health.md:67]: I agree. not sure we need this section

**Review by camicano — CHANGES_REQUESTED** [review]
@sharmoo let me know if you have any questions :)

**Comment by Deborah-Digges** [comment]
Closing after consulting with Eric Johson.

1) Will add health information to the main log streams page
2) Will be more succinct and try not to explain UI in too much detail
